src/common/options/rgw.yaml.in

```diff
@@ -1421,6 +1421,10 @@ options:
   level: advanced
   desc: Enable ops log
   fmt_desc: Enable logging for each successful Ceph Object Gateway operation.
+  long_desc: Log every Ceph Object Gateway operation. This option is very
+    verbose and may fill up the rgw.log pool quickly. Please use --debug-rgw
+    and --debug-ms command-line options for the Ceph Object Gateway nodes
+    instead.
```

Comment:
thanks @ktdreyer, i agree that we should give better guidance on how to use the feature. we know that the current design is not sustainable, because the logs just grow without bound

> This option is very verbose and may fill up the rgw.log pool quickly.

note that this part is specific to `rgw_ops_log_rados` (which defaults to true). the same issue comes up with file-based logging via `rgw_ops_log_file_path`

> Please use --debug-rgw and --debug-ms command-line options for the Ceph Object Gateway nodes instead.

i really don't think that debug logging is an acceptable replacement. first, the ops log may contain information that isn't written to rgw's debug logs. second, i never want to force users to scrape rgw's logs (with the sole exception of the frontend's access log line which we intentionally write in a common apache format)

ideally, we'd actually complete this feature with some kind of automated, policy-driven management

but for now, it's probably enough just to say that rgw does not do any cleanup of these log entries, so the admin MUST have their own process to manage and trim them

Reply:
The file-based logging doesn't really have the issue if proper log rotation configuration is used. It does work properly with default cephadm log rotation.

Reply:
@cbodley Could you write the paragraph that you want for `long_desc`?

The reason I mentioned `--debug-rgw` and `--debug-ms` is that we have to tell users some alternative to `rgw_enable_ops_log`, and I've seen other people recommending those two options elsewhere. What would you recommend?

Reply:
first, i would never recommend `rgw_ops_log_rados`. we've talked about removing it entirely, and i'd support doing that. but at the very least, it should not default to true, and it's description should recommend that it never be enabled unless the admin provides their own automation to clean up its logs

as @cfsnyder points out, the ops-log-to-file option `rgw_ops_log_file_path` does have sane default behavior, at least with cephadm, so i think it makes much more sense as the default backend for `rgw_enable_ops_log` instead of `rgw_ops_log_rados`

what do you think @mattbenjamin @cfsnyder?

Reply:
`rgw_ops_log_socket_path` could also make a sensible default, but first i'd want to make sure that we document how to use it, along with an example script to consume it. if someone is willing to write this stuff, we can make it the default

if `rgw_ops_log_file_path` doesn't need extra docs or instructions, it's probably a better default in the short term

Reply:
It is not sensible to make the newest thing, the default, and making it so short term, would then make for a change in behavior to revert to ops-log-socket.

Reply:
I guess my real preference for ops-log-socket is to emphasize, getting audit into directly into some other location--rather than into a file on the rgw host that most likely then needs to be scraped by something else

Reply:
I guess it's easy to show an example script consuming it

Reply:
ok, that makes sense

Reply:
assuming we're in agreement on this, i've proposed https://github.com/ceph/ceph/pull/45737 to change the default to file